packages/rest/src/rest.ts

```diff
@@ -332,7 +370,7 @@ function makeActionName(op: Operation, name: string | undefined): string {
 const actionsKey = createStateSymbol("actions");
 export function $action(context: DecoratorContext, entity: Operation, name?: string) {
   // Generate the action name and add it as an operation path segment
-  const action = makeActionName(entity, name);
+  const action = `!!${makeActionName(entity, name)}`;
```

Comment:
not he biggest fan of this magic `!!` string. Do we have a better way? If not maybe something more descriptive and always using the constant

Reply:
There is a better way... but it would require major changes to autoroute and existing decorators like `@segment` which I think would be even more breaking. Currently `@segment` can accept a single string, which was why `@collectionAction` was trying to combine its two segments into one (and exactly what prompted this bug).

The problem is that you cannot apply the separator until _all_ decorators have run. The sentinel is the simplest way to achieve that.

Another possibility I explored and reverted was having `@segment` accept `string | string[]` but that got complex quickly. Another possibility would be something like `string | ActionSegment` which I haven't tried but could be simpler to reason about.

The other issue with `segmentSeparator` is that if you have a url like `/foo/bar/baz:blah`, the only segment you actually want to customize is the last one. If you use `@segmentSeparator(":")` it would be silly to get `:foo:bar:baz:blah` so something seems conceptually wrong with segment separator, but it's used internally to construct things, including static part of ARM resource paths, so touching it is fraught with peril and so I opted to create a new, more specific decorator than try to "fix" the existing one.

Reply:
I could expose a helper method `getActionSentinel` to expose the constant.  I thought about changing `getAction` to return a tuple of the value and the sentinel, but this would be a breaking change. Exposing the constant via a method would mean that emitters could always safely call this method instead of hard coding to whatever constant we choose.  Thoughts?

Reply:
I think we could do that:
create a new symbol `const ActionSeperator = Symbol.for("@cadl-lang/rest.action-seprator")`
create a new method `setSegments(segments: Array<string | typeof ActionSeperator>)`
the `@segment` decorator just calls to that method(don't change it for now)

make the action call to `setSegemnts` passing the symbol and replacing the symbol when resolving the routes

Reply:
With my latest changes, I completely get rid of the sentinel. However, it exposes an additional decorator and method, and it is not fully backwards compatible. Please take a look at the latest commit @timotheeguerin and @daviwil.